module4-authentication-and-security/r1.1.1-authentication-persistence/README.md

```diff
@@ -0,0 +1,268 @@
+
+This can be used to persist authentication as well. This is often done when a user logs in and activates the **Remember Me** function. When login validation is carried out successfully, the server should also save a user copy to the session as seen in the previous lesson:
+
+```js
+req.session.user = user;
+```
+
+And when requesting protected resources, we can check if the request session that is tied to the request cookies, has the user or not. If it has the user as stored in our login logic, it means the request is authentic.
+
+> ⚠️ **Warning**: When using `express-session`, the default server-side session storage, `MemoryStore`, is purposely not designed for a production environment. It will leak memory under most conditions, does not scale past a single process, and is meant for debugging and developing.
+
+> 💡 **Tip**: `express-session` comes with lots of adapters to store sessions as plug-in libraries. Check out [this list](https://github.com/expressjs/session#compatible-session-stores).
+
+## **Stateless Session Storage**
+
+In stateless session storage, we don't store any temporary data on the server. Thus, the server is state-less. Instead, we issue a session payload in a token that encapsulates the data and sends it to the client for future use in subsequent requests. JWT is often, the way to carry out this process.
+
+JWT is an encoded string that holds **claims** supposed to authenticate and identify a request.
+
+```
+// Example JSON Web Token
+
+eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTYzMjk0MTc4NCwiZXhwIjoxNjMyOTQ1Mzg0fQ.MVnMhRVLYhaGqtZEX4AiEMrQwFWRvRLp6MATodg6E6A
+```
+
+This is a string that has 3 parts separated by a dot (.) and `base64` encoded. It can be decoded easily using any `base64` decoder.
+
+```js
+const token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTYzMjk0MTc4NCwiZXhwIjoxNjMyOTQ1Mzg0fQ.MVnMhRVLYhaGqtZEX4AiEMrQwFWRvRLp6MATodg6E6A';
+
+token.split('.').forEach(part => console.log(atob(part)))
+
+/* Output:
+{"typ":"JWT","alg":"HS256"}
+
+{"sub":"1234567890","name":"John Doe","admin":true,"iat":1632941784,"exp":1632945384}
+
+1YÌ�KbªÖD_"ÊÐÀU½éèÀ¡Ø:
+```
+
+The first part defines the _algorithm_ and the type. The second part is called the payload or _claims_, which identifies the user, issue time, expiry time, and other data. The third part is called the _signature_.
+
+When a user logs in, the server will issue this token by first defining the claims, signing them, finally encoding and sending them to the client. The client will persist this token somewhere in the browser, most commonly in the `localStorage`.
+
+To make an authenticated request by the client, it needs to pass this token to the server. Commonly, it is added to the request headers using the `Authorization` header.
+
+![JWT Diagram](../assets/use-of-jwt.webp)
+
+### **JWT Validation**
```

Comment:
no need to bold headers, also markdown formatter is saying that you might've messed up the code backticks after line 84 cuz the coloring is off

Reply:
thanks for catching this.